Source/Common/Include/MPIWrapper.h

```diff
@@ -304,6 +308,45 @@ class MPIWrapper : public std::enable_shared_from_this<MPIWrapper>
             fflush(stderr);
         }
         Ping("requestnodes (after change)");
+
+        // If all ranks run on a single host, we can enable optimized communication
+        // paths (e.g. NCCL). To determine if a single machine is being used, we
+        // check that MPI_Get_processor_name matches for all ranks.
+        const int nameMax = MPI_MAX_PROCESSOR_NAME + 1;
+        char myName[nameMax] = {0};
+        int  myNameLen = 0;
+        MPI_Get_processor_name(myName, &myNameLen) || MpiFail("requestnodes: MPI_Get_processor_name");
+        myName[myNameLen] = '\0';
+
+        std::vector<char> nameBuffer(m_numNodesInUse * nameMax);
+        char* allNames = nameBuffer.data();
+        MPI_Allgather(myName, nameMax, MPI_CHAR, allNames, nameMax, MPI_CHAR, m_currentComm)
+            || MpiFail("requestnodes: MPI_Allgather");
+
+        m_multiHost = false;
+        for(size_t i=1; i<m_numNodesInUse; i++)
+        {
+            if (strcmp(allNames, allNames+i*nameMax) != 0)
+            {
+                m_multiHost = true;
+                break;
+            }
+        }
+
+        if (!m_multiHost)
+        {
+            fprintf(stderr, "requestnodes [%s]: using %d out of %d MPI nodes (single machine) (%d requested); we (%d) are %s\n",
+                    msg, (int) m_numNodesInUse, (int) m_numMPINodes, (int) requestednodes,
+                    (int) CurrentNodeRank(), IsIdle() ? "out (idle)" : "in (participating)");
+        }
+        else
+        {
+            fprintf(stderr, "requestnodes [%s]: using %d out of %d MPI nodes (%d requested); we (%d) are %s\n",
+                    msg, (int) m_numNodesInUse, (int) m_numMPINodes, (int) requestednodes,
+                    (int) CurrentNodeRank(), IsIdle() ? "out (idle)" : "in (participating)");
+        }
```

Comment:
Maybe you can roll that into one, using another ternary operator. (Like already existing for the last %s)